dimagi/commcare-hq
# Make case tile widths dynamic

## Product Description
Case list tiles are currently set to a static 800px wide (or a fraction thereof). This change makes case tile widths dynamic based on window size.
Before:

https://github.com/dimagi/commcare-hq/assets/100609770/b61f954f-be5c-4c2d-a47e-6aa976f5e907

After:

https://github.com/dimagi/commcare-hq/assets/100609770/781070c9-8bd0-4bc7-8a32-285d4a829693

## Technical Summary
[USH-3137](https://dimagi-dev.atlassian.net/browse/USH-3137)
Arranges existing CSS grid layout of case tiles based on fractional units rather than static pixels, making the sizes dynamic by default. For case tiles shown > 1 wide, uses a wrapper to define the bounds of each tile's sizing, and adds an overall container to organize the tiles into a visible grid. Uses a newer CSS feature, container queries, to apply a custom xml attribute  `uniform-units`, which organizes a tile's content with grids of equal height and width.

Also changes `var` throughout `views.js` to `const` or `let` as appropriate for consistency/modernization.

## Feature Flag
`case_list_tile`, with some options under `case_list_custom_xml`

## Safety Assurance
Will only affect those using case tiles in web apps (very few if any at the moment), and only changes the UI visually. Does not change how data is loaded into the tiles or how they are interacted with to select cases.

### Safety story

### QA Plan
Will send to QA to ensure UI displays correctly and does not affect functionality.

### Rollback instructions


### Labels & Review


[USH-3137]: https://dimagi-dev.atlassian.net/browse/USH-3137?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
